From a9037a7f3cf95d5a77a070b86603f481e3e7df33 Mon Sep 17 00:00:00 2001
From: Denis <benato.denis96@gmail.com>
Date: Thu, 28 Sep 2023 03:51:03 +0200
Subject: [PATCH] debug

---
 drivers/platform/x86/asus-wmi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/x86/asus-wmi.c b/drivers/platform/x86/asus-wmi.c
index 8bef66a2f0ce..c06b5605f5d1 100644
--- a/drivers/platform/x86/asus-wmi.c
+++ b/drivers/platform/x86/asus-wmi.c
@@ -3346,8 +3346,12 @@ static void asus_wmi_notify(u32 value, void *context)
 			return;
 		}
 
-		if (code == WMI_EVENT_QUEUE_END || code == WMI_EVENT_MASK)
+		if (code == WMI_EVENT_QUEUE_END || code == WMI_EVENT_MASK) {
+			pr_warn("asus-wmi.c asus_wmi_notify: %d -- WMI_EVENT_QUEUE_END | WMI_EVENT_MASK \n", code);
 			return;
+		} else {
+			pr_warn("asus-wmi.c asus_wmi_notify: %d -- calling asus_wmi_handle_event_code \n", code);
+		}
 
 		asus_wmi_handle_event_code(code, asus);
 
-- 
2.42.0

