From a9037a7f3cf95d5a77a070b86603f481e3e7df33 Mon Sep 17 00:00:00 2001
From: Denis <benato.denis96@gmail.com>
Date: Thu, 28 Sep 2023 03:51:03 +0200
Subject: [PATCH 1/2] debug

---
 drivers/platform/x86/asus-wmi.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/platform/x86/asus-wmi.c b/drivers/platform/x86/asus-wmi.c
index 8bef66a2f0ce..c06b5605f5d1 100644
--- a/drivers/platform/x86/asus-wmi.c
+++ b/drivers/platform/x86/asus-wmi.c
@@ -3346,8 +3346,12 @@ static void asus_wmi_notify(u32 value, void *context)
 			return;
 		}
 
-		if (code == WMI_EVENT_QUEUE_END || code == WMI_EVENT_MASK)
+		if (code == WMI_EVENT_QUEUE_END || code == WMI_EVENT_MASK) {
+			pr_warn("asus-wmi.c asus_wmi_notify: %d -- WMI_EVENT_QUEUE_END | WMI_EVENT_MASK \n", code);
 			return;
+		} else {
+			pr_warn("asus-wmi.c asus_wmi_notify: %d -- calling asus_wmi_handle_event_code \n", code);
+		}
 
 		asus_wmi_handle_event_code(code, asus);
 
-- 
2.42.0


From a0cabb4ea86e44c28bcae75feaa106cf0f90122a Mon Sep 17 00:00:00 2001
From: Denis <benato.denis96@gmail.com>
Date: Thu, 28 Sep 2023 14:09:13 +0200
Subject: [PATCH 2/2] more wmi debug, added LED fixes for asus-hid

---
 drivers/hid/hid-asus.c          | 8 +++++++-
 drivers/hid/hid-ids.h           | 1 +
 drivers/platform/x86/asus-wmi.c | 2 ++
 3 files changed, 10 insertions(+), 1 deletion(-)

diff --git a/drivers/hid/hid-asus.c b/drivers/hid/hid-asus.c
index fd61dba88233..64b70bbfd0c5 100644
--- a/drivers/hid/hid-asus.c
+++ b/drivers/hid/hid-asus.c
@@ -897,7 +897,10 @@ static int asus_input_mapping(struct hid_device *hdev,
 		case 0xb3: asus_map_key_clear(KEY_PROG3);	break; /* Fn+Left next aura */
 		case 0x6a: asus_map_key_clear(KEY_F13);		break; /* Screenpad toggle */
 		case 0x4b: asus_map_key_clear(KEY_F14);		break; /* Arrows/Pg-Up/Dn toggle */
-
+		case 0xa5: asus_map_key_clear(KEY_F15);		break; /* ROG Ally left back */
+		case 0xa6: asus_map_key_clear(KEY_F16);		break; /* ROG Ally QAM button */
+		case 0xa7: asus_map_key_clear(KEY_F17);		break; /* ROG Ally ROG long-press */
+		case 0xa8: asus_map_key_clear(KEY_F18);		break; /* ROG Ally ROG long-press-release */
 
 		default:
 			/* ASUS lazily declares 256 usages, ignore the rest,
@@ -1258,6 +1261,9 @@ static const struct hid_device_id asus_devices[] = {
 	{ HID_USB_DEVICE(USB_VENDOR_ID_ASUSTEK,
 	    USB_DEVICE_ID_ASUSTEK_ROG_NKEY_KEYBOARD3),
 	  QUIRK_USE_KBD_BACKLIGHT | QUIRK_ROG_NKEY_KEYBOARD },
+	{ HID_USB_DEVICE(USB_VENDOR_ID_ASUSTEK,
+	    USB_DEVICE_ID_ASUSTEK_ROG_NKEY_ALLY),
+	  QUIRK_USE_KBD_BACKLIGHT | QUIRK_ROG_NKEY_KEYBOARD },
 	{ HID_USB_DEVICE(USB_VENDOR_ID_ASUSTEK,
 	    USB_DEVICE_ID_ASUSTEK_ROG_CLAYMORE_II_KEYBOARD),
 	  QUIRK_ROG_CLAYMORE_II_KEYBOARD },
diff --git a/drivers/hid/hid-ids.h b/drivers/hid/hid-ids.h
index 8a310f8ff20f..7c3a4788c921 100644
--- a/drivers/hid/hid-ids.h
+++ b/drivers/hid/hid-ids.h
@@ -208,6 +208,7 @@
 #define USB_DEVICE_ID_ASUSTEK_ROG_NKEY_KEYBOARD	0x1866
 #define USB_DEVICE_ID_ASUSTEK_ROG_NKEY_KEYBOARD2	0x19b6
 #define USB_DEVICE_ID_ASUSTEK_ROG_NKEY_KEYBOARD3	0x1a30
+#define USB_DEVICE_ID_ASUSTEK_ROG_NKEY_ALLY	0x1abe
 #define USB_DEVICE_ID_ASUSTEK_ROG_CLAYMORE_II_KEYBOARD	0x196b
 #define USB_DEVICE_ID_ASUSTEK_FX503VD_KEYBOARD	0x1869
 
diff --git a/drivers/platform/x86/asus-wmi.c b/drivers/platform/x86/asus-wmi.c
index c06b5605f5d1..87bc14ccd26d 100644
--- a/drivers/platform/x86/asus-wmi.c
+++ b/drivers/platform/x86/asus-wmi.c
@@ -3376,6 +3376,8 @@ static int asus_wmi_notify_queue_flush(struct asus_wmi *asus)
 		if (code < 0) {
 			pr_warn("Failed to get event during flush: %d\n", code);
 			return code;
+		} else {
+			pr_warn("asus-wmi.c asus_wmi_notify_queue_flush: %d\n", code);
 		}
 
 		if (code == WMI_EVENT_QUEUE_END || code == WMI_EVENT_MASK)
-- 
2.42.0

